lib/membrane_live_web/channels/event_channel.ex

```diff
@@ -98,12 +103,74 @@ defmodule MembraneLiveWeb.EventChannel do
     {:error, %{reason: "This link is wrong."}}
   end
 
+  defp create_event(socket, is_moderator) do
+    case :global.whereis_name(socket.assigns.event_id) do
+      :undefined ->
+        if is_moderator,
+          do: Event.start(socket.assigns.event_id, name: {:global, socket.assigns.event_id}),
+          else: {:error, :non_moderator}
+
+      pid ->
+        {:ok, pid}
+    end
+    |> case do
+      {:ok, event_pid} ->
+        Process.monitor(event_pid)
+        {:ok, Socket.assign(socket, %{event_pid: event_pid})}
+
+      {:error, {:already_started, event_pid}} ->
+        Process.monitor(event_pid)
+        {:ok, Socket.assign(socket, %{event_pid: event_pid})}
+
+      {:error, :non_moderator} ->
+        {:ok, Socket.assign(socket, %{event_pid: nil})}
+
+      {:error, reason} ->
+        Logger.error("""
+        Failed to start room.
+        Room: #{inspect(socket.assigns.event_id)}
+        Reason: #{inspect(reason)}
+        """)
+
+        {:error, %{reason: "failed to start event"}}
+    end
+  end
+
+  defp set_event_pid(socket) do
+    case :global.whereis_name(socket.assigns.event_id) do
+      :undefined -> socket
+      pid -> Socket.assign(socket, %{event_pid: pid})
+    end
+  end
+
+  defp join_event(socket) do
+    socket = set_event_pid(socket)
+
+    if is_nil(socket.assigns.event_pid),
+      do: raise("Event was not started before first joining attempt")
+
+    peer_id = "#{UUID.uuid4()}"
+
+    send(socket.assigns.event_pid, {:add_peer_channel, self(), peer_id})
+
+    {:ok, Socket.assign(socket, %{peer_id: peer_id})}
+  end
```

Comment:
I think we should move all private functions to the end of the file.

Reply:
Fixed.